### PR TITLE
Ports bbox-polygon package

### PR DESCRIPTION
**Pull request by armantorkzaban**

### CODE REVIEW

File: lib/bbox_polygon.dart [hunk 1]
```diff
@@ -0,0 +1,3 @@
+library turn_bbox_polygon.dart;
```
Comment (lukas-h) [inline lib/bbox_polygon.dart:1]: typo: it's `turf` instead of `turn`

File: lib/src/bbox_polygon.dart [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import '../helpers.dart';
```
Comment (lukas-h) [inline lib/src/bbox_polygon.dart:1]: please use a package: import

File: lib/src/bbox_polygon.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+/// Takes a bbox and returns an equivalent [Feature<Polygon>].
```
Comment (lukas-h) [inline lib/src/bbox_polygon.dart:3]: `Takes a [BBox]` would be correct, because the class is meant here

File: lib/src/bbox_polygon.dart [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+Feature<Polygon> bboxPolygon(BBox bbox,
+    {Map<String, dynamic> properties = const {}, dynamic id}) {
+  var west = bbox[0]!;
+  var south = bbox[1]!;
+  var east = bbox[2]!;
+  var north = bbox[3]!;
+
+  if (bbox.length == 6) {
+    throw Exception(
+        "@turf/bbox-polygon does not support BBox with 6 positions");
```
Comment (lukas-h) [inline lib/src/bbox_polygon.dart:19]: rename this to `bboxPolygon`, because this is the JS-spelling

File: lib/src/bbox_polygon.dart [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+        [
+          Position.of(lowLeft),
+          Position.of(lowRight),
+          Position.of(topRight),
+          Position.of(topLeft),
+          Position.of(lowLeft)
+        ]
+      ],
+    ),
+  );
```
Comment (lukas-h) [inline lib/src/bbox_polygon.dart:41]: let's set the bbox attr here, which should be bbox from the signature of the function, but cloned:
```dart
bbox: bbox.clone(),
```

File: test/components/bbox_polygon.dart [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+    },
+  );
+
+  test(
+    "bbox-polygon -- valid geojson",
+    () {
+      var poly = bboxPolygon(Bbox(0, 0, 10, 10));
+      var coordinates = poly.geometry!.coordinates;
+
+      expect((coordinates[0].length == 5), equals(true));
```
Comment (lukas-h) [inline test/components/bbox_polygon.dart:21]: you don't need the extra parentheses around the condition on the left side and also not the `equals(...)` surrounding the true on the right side

File: test/components/bbox_polygon.dart [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+  test(
+    "bbox-polygon -- Error handling",
+    () {
+      expect(() => bboxPolygon(Bbox(-110, 70, 5000, 50, 60, 3000)),
+          throwsA(isA<Exception>()));
+    },
+  );
+
+  test(
+    "bbox-polygon -- Translate BBox (Issue #1179)",
```
Comment (lukas-h) [inline test/components/bbox_polygon.dart:38]: I think, the issue reference is irrelevant for us here. WDYT?

**Review by lukas-h — CHANGES_REQUESTED** [review]
a few small improvements, generally good.
But please remove the `equals(...`, where you can. Because I don't think it's necessarry!

File: test/components/bbox_polygon.dart (hunk 7 above)
Comment (armantorkzaban) [reply test/components/bbox_polygon.dart]: It depends on how we port the packages. I would remove it and leave it to the future users to check for the details.

**Comment by lukas-h** [comment]
closed in favor of #99

**Pull request closed without merging**